---
title: "Transcriptomics and the analysis of RNA-Seq data"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Import from Lecture 14 
The data for this hands-on session comes from a published RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014).


```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
```

How many genes do we have?
```{r}
nrow(counts)
```

We have 4 replicates of control, and 4 of the drug condition. Let's calcualte thter average counts per gene for control samples and the drug samples. We;ll compare the averges. 

```{r}
#Which coloumsn are controls?
metadata$dex == "control"

#Now we pull out those columns.
control <- metadata[metadata$dex == "control", ]


#Now use the $id of the control data to acces the control columns in the counts and calucate the average. 
counts[, control$id]

control.mean<- rowSums(counts[, control$id]) /nrow(control)
names(control.mean) <-counts$ensgene

#Now the same for treated columns. 
treated <- metadata[metadata$dex == "treated", ]
counts[, treated$id]
treated.mean<- rowSums(counts[, treated$id]) /nrow(treated)
names(treated.mean) <-counts$ensgene

meancounts <- data.frame(control.mean, treated.mean)
meancounts

```
Now lets plot it!
```{r}
plot(meancounts)
#This plot looks messy, not spread out enough. Let's plot the log. 


plot(meancounts, log="xy")
#Looks better!

meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```


Removing genes we didn't get good data for. 
```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)
```
How many genes do I have left?
```{r}
nrow(mycounts)
```
Now, whats different between the treated and control? 
```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)

sum(up.ind)
sum(down.ind)
```

## Statistics with DESeq2 
Are our changes significant though? 

```{r}
library(DESeq2)

dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r}
plot(res$log2FoldChange, res$padj)
```


```{r}
plot(res$log2FoldChange, -log(res$padj), )
```

Lets make it colorful!
```{r}
mycols <- rep("gray", nrow(res))
mycols[ res$log2FoldChange > 2] <- "blue"
mycols[ res$log2FoldChange < -2] <- "blue"

plot(res$log2FoldChange, -log(res$padj), col=mycols)
abline(v=c(2, -2), lty=2)
abline(h=-log(0.05))
```







